lib/oli/delivery/metrics.ex

```diff
@@ -382,6 +431,256 @@ defmodule Oli.Delivery.Metrics do
     |> Enum.into(%{})
   end
 
+  @doc """
+  Calculates the learning mastery ("High", "Medium", "Low", "Not enough data")
+  for every learning objective of a given section
+
+    It returns a map:
+
+    %{objective_id_1 => "High",
+      ...
+      objective_id_n => "Low"
+    }
+  """
+  def mastery_per_learning_objective(section_slug) do
+    query =
+      from(sn in Snapshot,
+        join: s in Section,
+        on: sn.section_id == s.id,
+        where: sn.attempt_number == 1 and sn.part_attempt_number == 1 and s.slug == ^section_slug,
```

Comment:
Because the metric is "a percentage correct for **first attempts** on practice page part attempts that target a specific objective."
And also because the ticket explicitly required that clause 😄